store/sqlstore/retention_policy_store.go

```diff
@@ -0,0 +1,521 @@
+		FROM RetentionPolicies
+		LEFT JOIN RetentionPoliciesTeams ON RetentionPolicies.Id = RetentionPoliciesTeams.PolicyId
+		` + whereIdEqualsPolicyId + `
+		GROUP BY RetentionPolicies.Id
+		ORDER BY RetentionPolicies.DisplayName, RetentionPolicies.Id
+		LIMIT :Limit
+		OFFSET :Offset
+	) AS B ON RetentionPolicies.Id = B.Id
+	ORDER BY RetentionPolicies.DisplayName, RetentionPolicies.Id`
+	return
+}
+
+func (s *SqlRetentionPolicyStore) Get(id string) (*model.RetentionPolicyWithTeamAndChannelCounts, error) {
+	query, props := s.buildGetPolicyWithCountsQuery(id)
+	var policy model.RetentionPolicyWithTeamAndChannelCounts
+	if err := s.GetReplica().SelectOne(&policy, query, props); err != nil {
+		return nil, err
+	}
+	return &policy, nil
+}
+
+func (s *SqlRetentionPolicyStore) GetAll(offset, limit int) (policies []*model.RetentionPolicyWithTeamAndChannelCounts, err error) {
+	query, props := s.buildGetPoliciesWithCountsQuery("", offset, limit)
+	_, err = s.GetReplica().Select(&policies, query, props)
+	return
+}
+
+func (s *SqlRetentionPolicyStore) GetCount() (int64, error) {
+	return s.GetReplica().SelectInt("SELECT COUNT(*) FROM RetentionPolicies")
+}
+
+func (s *SqlRetentionPolicyStore) Delete(id string) error {
+	builder := s.getQueryBuilder().
+		Delete("RetentionPolicies").
+		Where(sq.Eq{"Id": id})
+	result, err := builder.RunWith(s.GetMaster()).Exec()
+	if err != nil {
+		return err
+	}
+	numRowsAffected, err := result.RowsAffected()
+	if err != nil {
+		return err
+	} else if numRowsAffected == 0 {
+		return errors.New("policy not found")
+	}
+	return nil
+}
+
+func (s *SqlRetentionPolicyStore) GetChannels(policyId string, offset, limit int) (channels []*model.Channel, err error) {
+	const query = `
+	SELECT Channels.* FROM RetentionPoliciesChannels
+	INNER JOIN Channels ON RetentionPoliciesChannels.ChannelId = Channels.Id
+	WHERE RetentionPoliciesChannels.PolicyId = :PolicyId
+	ORDER BY Channels.DisplayName, Channels.Id
+	LIMIT :Limit
+	OFFSET :Offset`
+	props := map[string]interface{}{"PolicyId": policyId, "Limit": limit, "Offset": offset}
+	_, err = s.GetReplica().Select(&channels, query, props)
+	return
+}
+
+func (s *SqlRetentionPolicyStore) AddChannels(policyId string, channelIds []string) error {
+	if len(channelIds) == 0 {
+		return nil
+	}
+	if err := s.checkChannelsExist(channelIds); err != nil {
+		return err
+	}
+	builder := s.getQueryBuilder().
+		Insert("RetentionPoliciesChannels").
+		Columns("policyId", "channelId")
+	for _, channelId := range channelIds {
+		builder = builder.Values(policyId, channelId)
+	}
+	_, err := builder.RunWith(s.GetMaster()).Exec()
+	if err != nil {
+		// check FK constraint violation
+		if pqErr, ok := err.(*pq.Error); ok && pqErr.Code == "23503" {
+			return store.NewErrNotFound("RetentionPolicy", policyId)
+		} else if mysqlErr, ok := err.(*mysql.MySQLError); ok && mysqlErr.Number == 0x5ac {
+			return store.NewErrNotFound("RetentionPolicy", policyId)
+		}
```

Comment:
Existing pattern is to use a Switch statement when determining error type. 
Also I would define error codes as constants. Easier to understand what they mean.
```
if err != nil {
	var pqErr *pq.Error
	var mysqlErr *mySql.MySQLError
	switch {
		case errors.As(err, &pqErr):
			if pqErr.Code == POSTGRES_FOREIGN_KEY_VIOLATION {
				return store.NewErrNotFound("RetentionPolicy", policyId)
			}
		...
		default:
			return err
		}

Reply:
Done.